validation/lst1-dl3-crab/make.py

```diff
@@ -0,0 +1,215 @@
+    SpectrumDatasetMaker,
+    WobbleRegionsFinder,
+    ReflectedRegionsBackgroundMaker,
+)
+from gammapy.maps import MapAxis, RegionGeom
+from gammapy.modeling import Fit
+from gammapy.modeling.models import (
+    SkyModel,
+    LogParabolaSpectralModel
+)
+from regions import PointSkyRegion
+import requests
+
+
+log = logging.getLogger(__name__)
+
+LST1_WORKING_DIR = Path(__file__).parent
+DL3_PATH = LST1_WORKING_DIR / "data"
+PATH_RESULTS = LST1_WORKING_DIR / "results"
+REFERENCE_RESULTS = LST1_WORKING_DIR / "reference"
+
+ZENODO_URL_LST1_CRAB = '11445184'
+
+
+@click.group()
+@click.option("--log-level", default="INFO", type=click.Choice(["DEBUG", "INFO", "WARNING"]))
+def cli(log_level):
+    logging.basicConfig(level=log_level)
+    get_data_from_zenodo()
+
+
+def dl3_to_dl4_reduction(dl3_path):
+    """DL3 to DL4 reduction. Returns a Datasets object."""
+    datastore = DataStore.from_dir(dl3_path)
+    observations = datastore.get_observations(required_irf="point-like")
+    target_position = SkyCoord.from_name("Crab Nebula", frame='icrs')
+    on_region = PointSkyRegion(target_position)
+
+    # True and estimated energy axes
+    energy_axis = MapAxis.from_energy_bounds(
+        0.01, 100, nbin=5, per_decade=True, unit="TeV", name="energy"
+    )
+    energy_axis_true = MapAxis.from_energy_bounds(
+        0.005, 200, nbin=10, per_decade=True, unit="TeV", name="energy_true"
+    )
+
+    geom = RegionGeom.create(region=on_region, axes=[energy_axis])
+
+    dataset_empty = SpectrumDataset.create(
+        geom=geom, energy_axis_true=energy_axis_true
+    )
+    dataset_maker = SpectrumDatasetMaker(
+        containment_correction=False,
+        selection=["counts", "exposure", "edisp"]
+    )
+    region_finder = WobbleRegionsFinder(n_off_regions=1)
+    bkg_maker = ReflectedRegionsBackgroundMaker(region_finder=region_finder)
+
+    datasets = Datasets()
+
+    for observation in observations:
+        dataset = dataset_maker.run(
+            dataset_empty.copy(name=str(observation.obs_id)), observation
+        )
+        dataset_on_off = bkg_maker.run(dataset, observation)
+        datasets.append(dataset_on_off)
+    
+    return datasets
+
+
+def spectral_model_fitting(datasets):
+    """Perform spectral model fitting.
+    
+    The analysis settings are the same used in the LST-1 performance
+    study, ApJ 956 80 (2023). It assumes a LogParabola spectral model,
+    with reference energy of 400 GeV, in the energy range from 50 GeV to 30 TeV.
+    The model is finally serialized as YAML file.
+    """
+    lp_spectral_model = LogParabolaSpectralModel(
+        amplitude=1e-12 * u.Unit("cm-2 s-1 TeV-1"),
+        alpha=2,
+        beta=0.1,
+        reference=0.4 * u.TeV,
+    )
+    lp_spectral_model.parameters["alpha"].min = 0
+    lp_spectral_model.parameters["beta"].min = 0
+
+    lp_model = SkyModel(spectral_model=lp_spectral_model, name="crab")
+
+    # Energy range for spectral model fitting
```

Comment:
Side note: this might be worth introducing as a helper function/method for `Datasets`